### PR TITLE
kubelet config should ignore file start with dots

### PR DESCRIPTION
**Pull request by resouer**
Fixes: #39156

Ignore files started with dot.

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot gce etcd3 e2e test this

File: pkg/kubelet/config/file_linux.go [hunk 1]
```diff
@@ -103,6 +104,10 @@ func (s *sourceFile) processEvent(e *inotify.Event) error {
 
 	switch eventType {
 	case podAdd, podModify:
+		// Ignore file start with dots
```
Comment (yujuhong) [inline pkg/kubelet/config/file_linux.go]: Shouldn't this be checked at the beginning of this function?

Comment (resouer) [inline pkg/kubelet/config/file_linux.go]: Oh, yes. Just pushed a new commit.

File: pkg/kubelet/config/file_linux.go [hunk 2]
```diff
@@ -79,6 +80,10 @@ func (s *sourceFile) watch() error {
 }
 
 func (s *sourceFile) processEvent(e *inotify.Event) error {
+	// Ignore file start with dots
+	if strings.HasPrefix(e.Name, ".") {
```
Comment (yujuhong) [inline pkg/kubelet/config/file_linux.go]: nit: log a message about this with V(4).

Comment (resouer) [inline pkg/kubelet/config/file_linux.go]: ack, and do you think we need to doc this behaviour in description of `--pod-manifest-path`?

**Comment by resouer** [comment]
@k8s-bot unit test this

**Comment by resouer** [comment]
@yujuhong rebased & updated to fix nits, please take a look.

File: cmd/kubelet/app/options/options.go [hunk 3]
```diff
@@ -91,7 +91,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringSliceVar(&s.APIServerList, "api-servers", []string{}, "List of Kubernetes API servers for publishing events, and reading pods and services. (ip:port), comma separated.")
 	fs.MarkDeprecated("api-servers", "Use --kubeconfig instead. Will be removed in a future version.")
 
-	fs.StringVar(&s.PodManifestPath, "pod-manifest-path", s.PodManifestPath, "Path to to the directory containing pod manifest files to run, or the path to a single pod manifest file.")
+	fs.StringVar(&s.PodManifestPath, "pod-manifest-path", s.PodManifestPath, "Path to to the directory containing pod manifest files to run, or the path to a single pod manifest file, file starts with dots will be ignored.")
```
Comment (yujuhong) [inline cmd/kubelet/app/options/options.go]: change `,` to `;`:  s/pod manifest file,/pod manifest file;

**Comment by dchen1107** [comment]
/approve

and looks good to me except above nits pointed by @yujuhong

File: cmd/kubelet/app/options/options.go (hunk 3 above)
Comment (resouer) [inline cmd/kubelet/app/options/options.go]: ack, fixed, thanks!

**Comment by resouer** [comment]
@k8s-bot non-cri e2e test this

**Comment by resouer** [comment]
@k8s-bot non-cri e2e test this

**Comment by resouer** [comment]
Thanks, all nits fixed.

**Pull request merged**